contracts/governance/utils/Votes.sol

```diff
@@ -49,30 +57,33 @@ abstract contract Votes is IVotes, Context, EIP712 {
     }
 
     /**
-     * @dev Returns the amount of votes that `account` had at the end of a past block (`blockNumber`).
+     * @dev Returns the amount of votes that `account` had at a specific moment in the past. If the `clock()` is
+     * configured to use block numbers, this will return the value the end of the corresponding block.
      *
      * Requirements:
      *
-     * - `blockNumber` must have been already mined
+     * - `timepoint` must be in the past. If operating using block numbers, the block must be already mined.
      */
-    function getPastVotes(address account, uint256 blockNumber) public view virtual override returns (uint256) {
-        return _delegateCheckpoints[account].getAtProbablyRecentBlock(blockNumber);
+    function getPastVotes(address account, uint256 timepoint) public view virtual override returns (uint256) {
+        require(timepoint < clock(), "Votes: future lookup");
+        return _delegateCheckpoints[account].upperLookup(SafeCast.toUint32(timepoint)); // optimistic ?
```

Comment:
What do you mean by the "optimistic" comment here?

Reply:
- `getAtProbablyRecentBlock` was optimistic.
- `upperLookup` is not.

The comment is a TODO, should we re-enforce optimistic lookups here (we don't have the right function available in Trace224).

Reply:
Oh you meant the "recenct" function variant. We should definitely use that. Would this be `upperLookupRecent`? Does that name make sense?

Reply:
Added